### PR TITLE
feat: add CLI arguments to output file name, other minor fixes and format

### PR DESCRIPTION
**Pull request by meagharty**
Some minor changes to this script for usability. I made these changes before deciding to use livebook for this investigation.

### CODE REVIEW

File: scripts/getArchive.py [hunk 1]
```diff
@@ -61,7 +67,7 @@ def matches_filters(ent, args):
 
 
 def entity_trip(ent):
-    if "trip_update" in ent:
+    if "trip_update" in ent and ent["trip_update"] is not None:
```
Comment (bklebe) [inline scripts/getArchive.py:70]: question: are we publishing null TripUpdates somewhere? I would expect the field to be missing rather than null where we don't provide it.

File: scripts/getArchive.py [hunk 2]
```diff
@@ -183,21 +194,37 @@ def main(args):
         args["stops"] = []
 
     if not args["output"]:
+        extra_filters = "-".join(
+            [
+                "{0}={1}".format(k, v)
+                for (k, v) in args.items()
+                if isinstance(v, str) and k != "feed" and k != "datetime"
```
Comment (bklebe) [inline scripts/getArchive.py:201]:
```suggestion
                if isinstance(v, str) and k not in {"feed", "datetime"}
```

File: README.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ Downloads archived MBTA real-time data feeds from various sources.
 ## Prerequisites
 
 * Python 3
-  * **Mac/Linux:** can install using [asdf](https://asdf-vm.com/) (`asdf install` in this directory)
+  * **Mac/Linux:** can install using [asdf](https://asdf-vm.com/) (`asdf install` in this directory) and then `pip install --user pipenv`
```
Comment (bklebe) [inline README.md:8]: suggestion: this doc change should probably be its own commit, it doesn't match the commit title. thank you for updating this, though!

**Review by beatrixada — DISMISSED** [review]

File: scripts/getArchive.py (hunk 2 above)
Comment (bklebe) [reply scripts/getArchive.py:201]: seems simpler to use a set literal here, up to you though.

File: README.md (hunk 3 above)
Comment (meagharty) [reply README.md]: https://github.com/mbta/prediction_loc/pull/32

File: scripts/getArchive.py (hunk 1 above)
Comment (meagharty) [reply scripts/getArchive.py:70]: We could create a rtr ticket for this. I'd rather not spend much more time on script itself though. We're better off spending that time on a livebook implementation. I have a work in progress one but it's not ready to be committed to this repository.

That said here's an example from the script of publishing an null TripUpdate within the VehiclePosition feed:

null TripUpdate within a VehiclePosition in rtr's enhanced vehicle position feed (*rtr_VehiclePositions_enhanced.json):
```
{
    "id": "1706048058_G-10186",
    "alert": None,
    "vehicle": {
        "position": {
            "speed": 5.516473599999999,
            "bearing": 45.0,
            "latitude": 42.32259,
            "longitude": -71.20525,
            "odometer": None,
        },
        "timestamp": 1706048058,
        "trip": {
            "start_time": "17:12:00",
            "start_date": "20240123",
            "trip_id": "60564884-20:15-BabckBackBayGovtGovtCtrKenmoreCDGovtCtrNorthStaGovtCtrNorthStaHeathNorthSusNorthMedfordNorthUnionSuspendD00",
            "route_id": "Green-D",
            "direction_id": 1,
            "schedule_relationship": "SCHEDULED",
            "revenue": True,
        },
        "stop_id": "70168",
        "vehicle": {
            "id": "G-10186",
            "label": "3677-3833",
            "consist": [{"label": "3677"}, {"label": "3833"}],
            "license_plate": None,
            "wheelchair_accessible": "NO_VALUE",
        },
        "multi_carriage_details": [
            {
                "id": None,
                "label": "3677",
                "occupancy_status": "NO_DATA_AVAILABLE",
                "occupancy_percentage": None,
                "carriage_sequence": 1,
            },
            {
                "id": None,
                "label": "3833",
                "occupancy_status": "NO_DATA_AVAILABLE",
                "occupancy_percentage": None,
                "carriage_sequence": 2,
            },
        ],
        "occupancy_status": None,
        "occupancy_percentage": None,
        "current_stop_sequence": 350,
        "current_status": "STOPPED_AT",
        "congestion_level": None,
    },
    "trip_update": None,
    "is_deleted": False,
    "shape": None,
}
```

`python -m pipenv run getArchive --datetime 2024-01-23T17:15:01 --feed subway_vehicle`

**Review by bfauble — APPROVED** [review]
